File: src/WebAPI/WebAPI/Models/Task.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebAPI.Models
{
    public class Task
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string StartTime { get; set; }
        public string EndTime { get; set; }
        public string Machine { get; set; }

        public long Duration { get; set; } // Unit : Minute
    }
}

File: src/algorithm-service/CustomSolver/src/main/java/wzy/jsp/simplest/demo/domain/communication/Task.java
package wzy.jsp.simplest.demo.domain.communication;

import wzy.jsp.simplest.demo.component.DateTimeConverter;

public class Task implements Comparable{
    public String Id;
    public String Name;
    public String StartTime;//"yyyy-MM-dd HH:mm:ss"
    public String EndTime;//"yyyy-MM-dd HH:mm:ss"
    public String Machine;
    public int Duration;//unit: minute

    @Override
    public int compareTo(Object o) {
        Task s = (Task) o;

        if (new DateTimeConverter().CalculateCountValueBetweenDates(this.StartTime,s.StartTime)<0) {
            return 1;
        }
        else {
            return -1;
        }
    }
}
